### PR TITLE
#264: disable incompatible mod warning

### LINKED ISSUE #264: Disable incompatible mod warning option
There's an checkbox to disable incompatible mod warnings in the TMPE mod options. As far as I can tell this seems mainly related to detection of methods overwritten by other mods (ie. the error will appear after starting/loading a game).

Since the recently added #91 incompatible mod checker (which displays warning at start up), the option is confusing some users - they are expecting it to hide the warning screen at start up, and are probably not aware of the one in-game.

Would it be worth changing it so that it toggles the start-up mod checker, and not the in-game method override checker?

Also, would the in-game override checker be affected by the huge refactor taking place as part of #119 harmony migration?

### LINKED ISSUE #284: Incompatible Detector Should Not Warn When Not Enabled
As much as I enjoy this new incompatible mod thing, it's annoying when I'm subscribed to a mod for testing purposes and, even if the mod's not enabled, it will warn me that TM:PE won't work with it anyways. TM:PE should warn me in the follow situations:

- If the state of TM:PE is toggled (on or off) ONLY for enabled mods.
- If the incompatible mod is enabled.

At least, this is my opinion.

### PR DESCRIPTION
**Pull request by krzychu124**
Resolves #264 and resolves #284

_Key features:_

- checkbox in options menu to turn off checking for incompatible mods
- checkbox at the bottom of `Incompatible mods detected` dialog to turn off checking at next startup or mod enable (does the same as option above)
- value is persisted in `TMPE_GlobalConfig.xml`

 Minor bug to solve before merging: **_[Fixed in latest commit]_**
~~- when incompatible mods detected, `Autorun at startup` checkbox state in the `Incompatible mods detected` dialog is not syncronized with corresponding value displayed in Mod options menu, but value change is properly saved to config.~~ 

_TODO:_
- [x] run incompatible mod checker on demand e.g. from Mod options menu (visible only when entering from game main menu)
- [x] implement #284 enhancement

### CODE REVIEW

**Comment by aubergine10** [comment]
Shall I add in the filter to only check enabled mods?

**Comment by krzychu124** [comment]
Updated description 😉

**Comment by krzychu124** [comment]
It looks like it is working as intended 😃

**Comment by aubergine10** [comment]
The text of these options is too similar IMO:

incompat

What about:

```
o Notify me if a mod conflict is detected
o Scan for known incompatible mods on startup
    o Ignore disabled mods
```

What do you think?

**Comment by krzychu124** [comment]
Good point. I like it 😃

**Comment by aubergine10** [comment]
Or instead of:

```
o Notify me if a new mod conflict is detected at any time
```

Maybe:

```
o Notify if there is an unexpected mod conflict
```

**Comment by aubergine10** [comment]
auto

Maybe change to:

```
o Scan for known incompatible mods on startup
```

This way it uses same locale string as in mod options, and also more closely ties together the checkbox on the pop-up with the checkbox in mod options?

**Comment by aubergine10** [comment]
The options all seem to work as intended btw. Both on start up and also when enabling/disabling the mod.

**Comment by krzychu124** [comment]
I will make all changes tomorrow 😉

**Comment by aubergine10** [comment]
Missing update to https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/blob/master/TLM/TLM/Resources/lang_template.txt which is used as basis of new localisations.

**Comment by krzychu124** [comment]
Good catch... I think something is broken because I can't see that file in project files view 😕 Alphabetically sorted:

Zrzut ekranu (156)

Anyway I'll update it 😃

**Comment by aubergine10** [comment]
Well, it shouldn't be included in the workshop package, which is why it's not in the solution.

I wonder if it's even needed, as we could just tell people to use the `lang.txt` as template instead?

I used the `lang_template.txt` in the localisation guide a that's what @VictorPhilipp had in the original wiki documentation for localisation. It's essentially a clone of `lang.txt` as far as I can tell.

**Comment by krzychu124** [comment]
I think that for translation better would be `lang.txt` because using just template you can't translate tutorial keys.

File: TLM/TLM/Resources/lang_zh.txt [hunk 1]
```diff
@@ -222,10 +222,13 @@ Compact_main_menu 紧凑主菜单
...
-Show_error_message_if_a_mod_incompatibility_is_detected 检测到MOD不兼容时提示
 Remove_parked_vehicles 移除停放的车辆
 Node_is_level_crossing 此节点级别交汇，不可禁用红绿灯。
 Experimental_features Experimental features
 Turn_on_red Turn on red
 Vehicles_may_turn_on_red 车辆红灯可转弯
-Also_apply_to_left/right_turns_between_one-way_streets Also apply to left & right turns between one-way streets
\ No newline at end of file
+Also_apply_to_left/right_turns_between_one-way_streets Also apply to left & right turns between one-way streets
+Scan_for_known_incompatible_mods_on_startup Scan for known incompatible mods on startup
```
Comment (Emphasia) [inline TLM/TLM/Resources/lang_zh.txt]:
```suggestion
Scan_for_known_incompatible_mods_on_startup 启动时检测是否存在已知不兼容MOD
```

File: TLM/TLM/Resources/lang_zh.txt [hunk 2]
```diff
@@ -222,10 +222,13 @@ Compact_main_menu 紧凑主菜单
...
+Ignore_disabled_mods Ignore disabled mods
```
Comment (Emphasia) [inline TLM/TLM/Resources/lang_zh.txt]:
```suggestion
Ignore_disabled_mods 忽略未启用MOD
```

File: TLM/TLM/Resources/lang_zh.txt [hunk 3]
```diff
@@ -222,10 +222,13 @@ Compact_main_menu 紧凑主菜单
...
+Traffic_Manager_detected_incompatible_mods Traffic Manager detected incompatible mods
```
Comment (Emphasia) [inline TLM/TLM/Resources/lang_zh.txt]:
```suggestion
Traffic_Manager_detected_incompatible_mods 检测到的不兼容MOD
```

File: TLM/TLM/Resources/lang_zh.txt [hunk 4]
```diff
@@ -222,10 +222,13 @@ Compact_main_menu 紧凑主菜单
...
+Notify_me_if_there_is_an_unexpected_mod_conflict 检测到MOD不兼容时提示
```
Comment (Emphasia) [inline TLM/TLM/Resources/lang_zh.txt]:
```suggestion
Notify_me_if_there_is_an_unexpected_mod_conflict 检测到不兼容MOD时提示
```

**Comment by aubergine10** [comment]
@krzychu124 I've updated the localisation page in wiki to state `lang.txt` should be used as template.

**Comment by aubergine10** [comment]
@FireController1847 Can you confirm the skipping disabled mods thing works as you expect?

**Comment by krzychu124** [comment]
I've updated translations from @Emphasia review and included those from PR in @VictorPhilipp repo as well

**Comment by aubergine10** [comment]
LGTM! Ok to merge?

**Comment by aubergine10** [comment]
I agree with @FireController1847 , having the 'ignore disabled' be active by default would be better.

Good catch @FireController1847 !

**Comment by krzychu124** [comment]
Changed 😉

**Pull request merged**